File: result/translate/transcoder/java2python/python/PERMUTE_TWO_ARRAYS_SUM_EVERY_PAIR_GREATER_EQUAL_K.py
def f_gold ( a , b , n , k ) :
    a = np.array ( a )
    b = np.array ( b )
    a = np.sort ( a , key = lambda x : x [ 0 ] )
    b = np.sort ( b , key = lambda x : x [ 0 ] )
    for i in range ( n ) :
        if a [ i ] + b [ i ] < k :
            return False
    return True

File: result/translate/chatgpt/java2cpp/c++/PERMUTE_TWO_ARRAYS_SUM_EVERY_PAIR_GREATER_EQUAL_K.cpp
bool f_gold(int a[], int b[], int n, int k) {
    int *a_copy = new int[n];
    int *b_copy = new int[n];
    
    std::copy(a, a + n, a_copy);
    std::copy(b, b + n, b_copy);
    
    std::sort(a_copy, a_copy + n, std::greater<int>());
    std::sort(b_copy, b_copy + n);
    
    for(int i = 0; i < n; i++) {
        if(a_copy[i] + b_copy[i] < k) {
            delete[] a_copy;
            delete[] b_copy;
            return false;
        }
    }
    
    delete[] a_copy;
    delete[] b_copy;
    return true;
}